internal/state/graph.go

```diff
@@ -123,46 +178,75 @@ func bindHTTPRouteToListeners(
 	ignored = true
 
 	for _, p := range ghr.Spec.ParentRefs {
-		// Step 1 - Ensure the ref references the Gateway and has a non-empty section name
-
-		if ignoreParentRef(p, ghr.Namespace, gwNsName) {
+		// FIXME(pleshakov) Support empty section name
+		if p.SectionName == nil || *p.SectionName == "" {
 			continue
 		}
 
-		ignored = false
+		// if the namespace is missing, assume the namespace of the HTTPRoute
+		ns := ghr.Namespace
+		if p.Namespace != nil {
+			ns = string(*p.Namespace)
+		}
 
 		name := string(*p.SectionName)
 
-		// Step 2 - Find a listener
-
-		// FIXME(pleshakov)
-		// For now, let's do simple matching.
-		// However, we need to also support wildcard matching.
-		// More over, we need to handle cases when a Route host matches multiple HTTP listeners on the same port when
-		// sectionName is empty and only choose one listener.
-		// For example:
-		// - Route with host foo.example.com;
-		// - listener 1 for port 80 with hostname foo.example.com
-		// - listener 2 for port 80 with hostname *.example.com;
-		// In this case, the Route host foo.example.com should choose listener 2, as it is a more specific match.
-
-		l, exists := listeners[name]
-		if !exists {
-			r.InvalidSectionNameRefs[name] = struct{}{}
+		// Below we will figure out what Gateway resource the parentRef references and act accordingly. There are 3 cases.
+
+		// Case 1: the parentRef references the winning Gateway.
+
+		if gw != nil && gw.Namespace == ns && gw.Name == string(p.Name) {
+
+			// Find a listener
+
+			// FIXME(pleshakov)
+			// For now, let's do simple matching.
+			// However, we need to also support wildcard matching.
+			// More over, we need to handle cases when a Route host matches multiple HTTP listeners on the same port when
+			// sectionName is empty and only choose one listener.
+			// For example:
+			// - Route with host foo.example.com;
+			// - listener 1 for port 80 with hostname foo.example.com
+			// - listener 2 for port 80 with hostname *.example.com;
+			// In this case, the Route host foo.example.com should choose listener 2, as it is a more specific match.
```

Comment:
Maybe I'm reading this wrong, but isn't Listener 1 more specific, both `foo.example.com`?

Reply:
yep, you're right